testing/tests/DevExpress.ui.widgets/listParts/commonTests.js

```diff
@@ -4030,3 +4031,81 @@ if(devices.real().deviceType === 'desktop') {
         });
     });
 }
+
+if(QUnit.urlParams['nojquery']) {
+    QUnit.module('ShadowDOM', {
+    }, () => {
+        QUnit.test('ShadowDom: drag item', function(assert) {
+            const root = document.querySelector('#list');
+            const container = document.createElement('div');
+            const list = document.createElement('div');
+
+            root.attachShadow({ mode: 'open' });
+            container.appendChild(list);
+            root.shadowRoot.appendChild(container);
+
+            const $list = $(list).dxList({
+                items: ['One', 'Two', 'Three'],
+                itemDragging: { allowReordering: true },
+                focusStateEnabled: true,
+            });
+
+            const $items = $list.find(toSelector(LIST_ITEM_CLASS));
+
+            const pointer = reorderingPointerMock($items.first());
+
+            pointer.dragStart().drag(34);
+            pointer.dragEnd();
+
+            const orderedItems = $list.find(toSelector(LIST_ITEM_CLASS)).toArray().map(e => e.innerText.trim());
+
+            assert.deepEqual(orderedItems, ['Two', 'Three', 'One']);
+
+            $(container).empty();
+        });
+
+        QUnit.test('ShadowDom: focus item', function(assert) {
+            const done = assert.async();
+
+            const root = document.querySelector('#list');
+            const list = document.createElement('div');
+            const container = document.createElement('div');
+
+            root.attachShadow({ mode: 'open' });
+            root.shadowRoot.appendChild(container);
+            container.appendChild(list);
+
+            const createEvent = eventName => {
+                const event = $.Event(eventName);
+
+                event.originalEvent = {
+                    type: eventName,
+                    target: { shadowRoot: root },
+                    path: [ $items.eq(1)[0] ],
+                    changedTouches: [{}]
+                };
+
+                return event;
+            };
+
+            const $list = $(list).dxList({
+                items: ['One', 'Two', 'Three'],
+                focusStateEnabled: true,
+            });
+
+            const $items = $list.find(toSelector(LIST_ITEM_CLASS));
+
+            $(root).trigger(createEvent('mousedown'));
+            $(root).trigger(createEvent('touchstart'));
+
+            setTimeout(function() {
+                assert.ok($items.eq(1).hasClass('dx-state-focused'));
```

Comment:
why do we need this settimeout here?

Reply:
it looks like the settings `dx-state-focused` className is asynchronous, so we need this timeout to check the className

Reply:
but it's not usually async. You can check neighbour tests for focus